module-system/sov-modules-macros/src/dispatch/derive_rpc.rs

```diff
@@ -371,17 +371,32 @@ pub(crate) fn rpc_outer_impls(
 
     let mut output_tokens = proc_macro2::TokenStream::new();
 
+    let mut merge_operations = proc_macro2::TokenStream::new();
+
     let types = match args {
         syn::Type::Tuple(tuple) => tuple.elems,
         _ => panic!("Expected a tuple of types"),
     };
 
+    let rpc_storage_struct = quote! {
+        #[derive(Clone)]
+        pub struct RpcStorage<C: Context> {
+            pub storage: C::Storage
+        }
+    };
+
     let original_impl = quote! {
         #(#attrs)*
         #input
+        #rpc_storage_struct
     };
+
     output_tokens.extend(original_impl);
 
+    // will be replaced in the below loop
```

Comment:
Shall we add a TODO? or are we planning to fix this hack before merging?

Reply:
yeah, i can add a todo for this - it's a bit complex because we let each module specify its own context, but the runner has only a single context and storage. we might need to reconcile that later